### ISSUE TITLE
LowQual on REF base.

### ISSUE THREAD

**Issue by alexjironkin** [phe-bioinformatics/PHEnix#9]
GATK does a funny call on some bases where there is only REF e.g.:

```
reference       32      .       A       .       24.25   LowQual AN=2;DP=6;MQ=60.0;MQ0=0 GT:DP   0/0:6
```

This seems to be due to low coverage and an error in a read. GATK sensibly says it **LowQual** variant, actually not reporting variant. Currently, we will filter this base as failed and in `vcf2fasta` it will be treated accordingly probably will stick **N**).

So the question is: should we be filtering out when _only_ **LowQual** filter fails? All it really does, is says that _variant_ is of low quality, not REF.

**Comment by antunderwood** [comment]
I would say we have to filter out and call N since we have none of the other metrics e.g AD by which to judge this position

**Comment by alexjironkin** [comment]
**LowQual** refers to low quality of an _ALT_ base, not _REF_ base.

https://samtools.github.io/hts-specs/VCFv4.2.pdf

**Comment by antunderwood** [comment]
Yes I get that but IMO it's a base that we can't say for sure what it is. In my mind whatever we put in the fasta it can't be a single base. Therefore for tree drawing what we put is somewhat arbitrary because of the way most ml algorithms interpret ambiguous bases

**Comment by richardemyers** [comment]
According to the samtools documentation referenced above:

"QUAL - quality: Phred-scaled quality score for the assertion made in ALT. i.e. −10log10 prob(call in ALT is wrong). If ALT is ‘.’ (no variant) then this is −10log10 prob(variant), and if ALT is not ‘.’ this is −10log10 prob(no variant). If unknown, the missing value should be specified. (Numeric)"

As I read this - low qual, where ALT is ".", means that there is an increased probability of a variant being present at that position, even though no ALT has been detected/reported. I'm not sure how you can establish that, other than low read depth. However, increased probability that a variant is present has the net effect of reducing the confidence of a ref only prediction at this site. If my interpretation is correct, I would agree with Anthony that the position should be ambiguous (gap or N specified by the user #10)

**Comment by richardemyers** [comment]
sorry - didn't mean to close it

**Comment by alexjironkin** [comment]
In the case above: ALT is "." and `-10log10 24.25` is 0 (for all intents and purposes). So probability of a variant is 0. How does this decreases confidence in REF base (if coverage is > `min_depth`)?

UPDATE: I did my math wrong: for ALT = "." the formula is `-10log10 Prob(var) = QUAL` doing some rearrangement we get `Prob(var) = 10^(QUAL/-10)` so for the above example Prob(var) at that position is `0.003758374` i.e. very low

**Comment by alexjironkin** [comment]
I have previously raised an issue with samtools about this: https://github.com/samtools/samtools/issues/566

**Comment by alexjironkin** [comment]
So Heng Li in bcftools thread reckons that LowQual positions should be filtered for ALT = "." as well as ALT != ".". I am not sure I agree with this if this assertion is used to generate an N in FASTA. If any thing it should be excluded from the analysis.
